hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/java/org/apache/hadoop/fs/http/client/HttpFSFileSystem.java

```diff
@@ -1756,6 +1762,18 @@ public FileStatus getFileLinkStatus(final Path path) throws IOException {
     return status.makeQualified(getUri(), path);
   }
 
+  @Override
+  public FsStatus getStatus(final Path path) throws IOException {
+    Map<String, String> params = new HashMap<>();
```

Comment:
Do we need to verify the path? Is it possible that path is null?

Reply:
Path can be null. Please refer to the method for more information
Filesystem#getStatus
```
/**
   * Returns a status object describing the use and capacity of the
   * filesystem. If the filesystem has multiple partitions, the
   * use and capacity of the partition pointed to by the specified
   * path is reflected.
   * @param p Path for which status should be obtained. null means
   * the default partition.
   * @return a FsStatus object
   * @throws IOException
   *           see specific implementation
   */
  public FsStatus getStatus(Path p) throws IOException {
    return new FsStatus(Long.MAX_VALUE, 0, Long.MAX_VALUE);
  }
```